### DISCUSSION TITLE
Docsearch doesn't work

### DISCUSSION THREAD

**Discussion by jcubic** [facebook/docusaurus#10007] [Q&A] [answered]
I've setup docSearch the same code works in DocSearch demo but fail in Docusaurus

Docusaurus
* https://stackblitz.com/edit/github-vefg3h?file=docusaurus.config.js

DocSearch basic example:
* https://codesandbox.io/p/sandbox/crazy-varahamihira-9d68rt?file=%2Fapp.js

I was reading different issues related to DocSearch and not of the things worked. I restarted the build multiple times, deleted the index, modified the config and nothing works.

**Comment by slorber** [comment, marked as answer]
**EDIT**: see Troubleshooting section added to our docs here:

https://docusaurus.io/docs/search#algolia-troubleshooting

---

Your index exists, but is not properly configured.

We are unable to query on these fields probably because they do not exist:

```
facetFilters: ["language:en",["docusaurus_tag:default","docusaurus_tag:docs-default-current"]]"}]}
```

If you disable contextual search (not recommended because it will disable that smart query), we can get results:

![CleanShot 2024-04-05 at 14 50 11](https://github.com/facebook/docusaurus/assets/749374/728e2299-ecc9-4446-ad97-a2b1cb0e21ae)


---


If you don't have these fields in your index, it is probably a configuration problem. 

![CleanShot 2024-04-05 at 14 53 09](https://github.com/facebook/docusaurus/assets/749374/f88617ee-0308-4f75-b725-83fdbf9bf2f9)


Verify your crawler config, and you might need to delete/recreate your index according to what other users reported, including the Algolia team (https://github.com/facebook/docusaurus/issues/9200#issuecomment-1667338492)

If you have index problems, you should contact the Algolia DocSearch support directly on their Discord server, because it's not a Docusaurus issue.

**Reply by jcubic** [reply]
Thanks, it seems to work, do you know how I should split single record into multiple record? Since it was indexed but doesn't work.

I used this based on Algolia documentation:

```javascript
      recordExtractor: ({ url, $, helpers }) => {
        if (url.pathname.match(/\/reference/)) {
          const baseRecord = {
            url,
            title: $("head title").text().trim(),
          };
          const records = helpers.splitContentIntoRecords({
            baseRecord,
            $elements: $("main"),
            maxRecordBytes: 1000,
            textAttributeName: "text",
            orderingAttributeName: "part",
          });
          return records;
        }
        // ... defaults
      }

```

The rest is the default from DocSearch docs.

I can ask a different question, so it's easier to find.

**Reply by slorber** [reply]
I'm not particularly skilled into customizing the Algolia crawler. If you want to use a custom config please contact the Algolia team directly, I can't help on that.

**Reply by jcubic** [reply]
ok, thanks
